client/my-sites/plugins/marketplace/marketplace-domain-upsell/index.tsx

```diff
@@ -0,0 +1,206 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect, useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import page from 'page';
+import DomainPicker from '@automattic/domain-picker';
+import { useShoppingCart } from '@automattic/shopping-cart';
+import { ThemeProvider } from 'emotion-theming';
+import styled from '@emotion/styled';
+import { translate } from 'i18n-calypso';
+import classnames from 'classnames';
+import { isDesktop } from '@automattic/viewport';
+import type { ResponseCart } from '@automattic/shopping-cart';
+import type { DomainSuggestions } from '@automattic/data-stores';
+
+/**
+ * Internal dependencies
+ */
+import CalypsoShoppingCartProvider from 'calypso/my-sites/checkout/calypso-shopping-cart-provider';
+import { domainRegistration } from 'calypso/lib/cart-values/cart-items';
+import {
+	MARKETPLACE_FLOW_ID,
+	ANALYTICS_UI_LOCATION_MARKETPLACE_DOMAIN_SELECTION,
+} from 'calypso/my-sites/plugins/marketplace/constants';
+import { getWpComDomainBySiteId } from 'calypso/state/sites/domains/selectors';
+import { fetchSiteDomains } from 'calypso/state/sites/domains/actions';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import { getProductsList, isProductsListFetching } from 'calypso/state/products-list/selectors';
+import { fillInSingleCartItemAttributes } from 'calypso/lib/cart-values';
+import Masterbar from 'calypso/layout/masterbar/masterbar';
+import Item from 'calypso/layout/masterbar/item';
+import getPreviousPath from 'calypso/state/selectors/get-previous-path';
+import { HorizontalRule } from '@wordpress/components';
+import ExternalLink from 'calypso/components/external-link';
+import MarketplaceShoppingCart from 'calypso/my-sites/plugins/marketplace/components/marketplace-shopping-cart';
+import theme from 'calypso/my-sites/plugins/marketplace';
```

Comment:
I'd call the themes module `/theme.ts` instead of `/index.tsx`. Naming it `index` suggests that it's an entrypoint/homepage of the module, but it isn't.

Also, what do you think about importing related modules with relative paths, i.e., `from '..'`? With that, the `plugins/marketplace` folder can be moved around to different location and all imports will still work.

Reply:
I just included the theme stuff in sort of a global place so that anyone within the module can access it from one place. But theme.ts does make more sense.

> Also, what do you think about importing related modules with relative paths, i.e., from '..'? With that, the plugins/marketplace folder can be moved around to different location and all imports will still work.

But doesn't calypso/... paths serve the same purpose but in even a better way? This allows us the flexibility to move any file/folder which is even within the module to any place outside of the module. 🤔 
Even within the module we can move files across multiple levels of folders without worrying about whether the IDE does the proper route refactors.

Reply:
Is there a downside to using the global absolute path? Build time or run time? curious to learn :)

Reply:
When using `calypso/` paths, they'll need all to be changed when moving the folder. Relative paths wouldn't need to be changed at all.

But it's OK to keep `calypso/`, apparently it's a personal preference that differs among folks.

Reply:
Oh ok I just understood 🤔  I am fine with this right now but will keep that in mind 👍